src/control/BarrettFingerKinematicSimulationPositionCommandExecutor.cpp

```diff
@@ -185,13 +209,26 @@ void BarrettFingerPositionCommandExecutor::step(double _timeSincePreviousCall)
   }
 
   mProximalDof->setPosition(newProximal);
-
   if (proximalGoalReached)
   {
     terminate();
     return;
   }
-}
 
 }
+
+//=============================================================================
+bool BarrettFingerKinematicSimulationPositionCommandExecutor::setCollideWith(
+  ::dart::collision::CollisionGroupPtr collideWith)
+{
+  std::lock_guard<std::mutex> lockSpin(mMutex);
+
```

Comment:
Nit: Check that `collideWith->getCollisionDetector()` is correct. Alternatively, update `mCollisionDetector` with that value.